### ISSUE TITLE
[Bug] `@beartype` raises `TypeError` when decorating immutable types

### ISSUE THREAD

**Issue by RomainBrault** [beartype/beartype#264]
I think it is impossible to use @beartype while inheriting from an immutable type (such as str) and a python type.

For example with str and Enum:

```python
"""MVC where beartype raises error with StrEnum **or** (str, enum)."""

import enum
import beartype


@beartype.beartype
class MyEnum(str, enum.Enum):
    """Beartype should not raise an errror."""

    A = "A"


if __name__ == "__main__":
    print(MyEnum("A"))
```

I get:

```bash
> python test.py
Traceback (most recent call last):
  File "/XXX/test.py", line 7, in <module>
    @beartype.beartype
     ^^^^^^^^^^^^^^^^^
  File "/XXX/lib/python3.11/site-packages/beartype/_decor/_cache/cachedecor.py", line 77, in beartype
    return beartype_object(obj, conf)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/XXX/lib/python3.11/site-packages/beartype/_decor/decorcore.py", line 97, in beartype_object
    _beartype_object_fatal(obj, conf, **kwargs)
  File "/XXX/lib/python3.11/site-packages/beartype/_decor/decorcore.py", line 172, in _beartype_object_fatal
    return _beartype_type(  # type: ignore[return-value]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/XXX/lib/python3.11/site-packages/beartype/_decor/decorcore.py", line 511, in _beartype_type
    attr_value_beartyped = beartype_object(
                           ^^^^^^^^^^^^^^^^
  File "/XXX/lib/python3.11/site-packages/beartype/_decor/decorcore.py", line 97, in beartype_object
    _beartype_object_fatal(obj, conf, **kwargs)
  File "/XXX/lib/python3.11/site-packages/beartype/_decor/decorcore.py", line 172, in _beartype_object_fatal
    return _beartype_type(  # type: ignore[return-value]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/XXX/lib/python3.11/site-packages/beartype/_decor/decorcore.py", line 536, in _beartype_type
    setattr(cls, attr_name, attr_value_beartyped)
TypeError: cannot set 'maketrans' attribute of immutable type 'str'
```

Because Beartype is trying to use setattr on an immutable type. Note that Python 3.11 StrEnum is also affected.

Prior to beartype 0.15 I just skipped the decorator on the enum class, but now with the import hooks it generates an annoying warning. I guess I could disable it, but maybe there's a better way to deal with this problem.

Thanks for your help.

**Comment by JWCS** [comment]
In the short term, you could decorate enum classes with `@typing.no_type_check`; that's the easiest one-off @beartype noop.
_(Read the updated FAQ on no-ops 2min ago)_

**Comment by leycec** [comment]
**Resolved by ded837868e.** Thanks so much for this *awesomely* detailed issue and minimal example, @RomainBrault. @beartype has got your back! Hopefully not literally, because... you need your back. :sweat_smile: 

In short, @beartype will support immutable types out-of-the-box in our upcoming @beartype 0.15.1 release – to be released as soon as I crush all of the *other* hideous issues that are currently assaulting our issue tracker. Let's pretend that will be fast.
